src/Engine/UnitActionLifecycles/UnloadLifecycle.java

```diff
@@ -30,28 +30,32 @@ public static class UnloadFactory extends UnitActionFactory
     public GameActionSet getPossibleActions(GameMap map, GamePath movePath, Unit actor, boolean ignoreResident)
     {
       XYCoord moveLocation = movePath.getEndCoord();
-      if( ignoreResident || map.isLocationEmpty(actor, moveLocation) )
-      {
-        if( actor.heldUnits.size() > 0 )
-        {
-          ArrayList<GameAction> unloadActions = new ArrayList<GameAction>();
+      Unit resident = map.getLocation(moveLocation).getResident();
 
-          // TODO: Consider using ignoreResident for dropoff points as well
-          for( Unit cargo : actor.heldUnits )
-          {
-            ArrayList<XYCoord> dropoffLocations = Utils.findUnloadLocations(map, actor, moveLocation, cargo);
-            for( XYCoord loc : dropoffLocations )
-            {
-              unloadActions.add(new UnloadAction(actor, movePath, cargo, loc));
-            }
-          }
+      if( !ignoreResident && null != resident )
+        return null;
+      if( actor.heldUnits.size() == 0 )
+        return null;
+      if( actor.model.unloadExclusionTerrain.contains(map.getEnvironment(moveLocation).terrainType) )
+        return null;
 
-          if( !unloadActions.isEmpty() )
-          {
-            return new GameActionSet(unloadActions);
-          }
+      ArrayList<GameAction> unloadActions = new ArrayList<GameAction>();
+
+      // TODO: Consider using ignoreResident for dropoff points as well
```

Comment:
Only if the space is in fog of war I'd think, but you already handled that case. Why use `ignoreResident` for dropoff points otherwise?

Reply:
AI logic - "I can unload this infantry on that mountain if I move my other infantry that's already there"

Reply:
Ah, right.